ui: Sortable loading state

vladlos opened the pull request:
Added global loading state to sort tables
Added loading state to database tables

![loading-database](https://user-images.githubusercontent.com/12850886/78561401-4e875500-7820-11ea-8462-53bd7310adf0.png)

Resolves: #46568

Release justification: low risk, high benefit changes to existing functionality

Release note (ui): database loading state design updates

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/46857)
<!-- Reviewable:end -->

vladlos committed:
ui: Sortable loading state

Added global loading state to sort tables
Added loading state to database tables

Resolves: #46568

Release justification: low risk, high benefit changes to existing functionality

Release note (ui): database loading state design updates

dhartunian reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46857)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @elkmaster)

___
*[pkg/ui/src/views/databases/containers/databaseTables/index.tsx, line 35 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/46857#-M4Q9U_50TuuSGncPyIH:-M4Q9U_50TuuSGncPyII:bb2fs3m) ([raw file](https://github.com/cockroachdb/cockroach/blob/51e8960053a8224856234ffc1fb610c3843f7d7d/pkg/ui/src/views/databases/containers/databaseTables/index.tsx#L35)):*
> ```XML
> class DatabaseTableListSortedTable extends SortedTable<TableInfo> {}
> 
> class DatabaseTableListEmpty extends React.Component {
> ```

it looks like the existing message of "This database has no tables" is removed with this PR. Can we keep that in for databases that in fact, have no tables.
___
*[pkg/ui/src/views/databases/containers/databaseTables/index.tsx, line 64 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/46857#-M4Q9VzGAU_vvlcrWBLh:-M4Q9VzGAU_vvlcrWBLi:b843umb) ([raw file](https://github.com/cockroachdb/cockroach/blob/280fa5e7e562e2ae1a425d9030b3e1f54c188979/pkg/ui/src/views/databases/containers/databaseTables/index.tsx#L64)):*
> ```XML
>                   sortSetting={sortSetting}
>                   onChangeSortSetting={(setting) => this.props.setSort(setting)}
>                   loading={!tableInfos}
> ```

Did you happen to see if we could access the `inFlight` property on the `cachedData` objects? Will this show as "loading" if the response is empty?


<!-- Sent from Reviewable.io -->

vladlos committed:
ui: Database loading state

returned empty state for database tables

used `isFlight` for Database table loading state

Release note (ui): database loading state design updates

dhartunian reviewed (changes requested):
> Reviewed 4 of 6 files at r1, 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46857)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dhartunian and @elkmaster)

___
*[pkg/ui/src/views/databases/containers/databaseSummary/index.tsx, line 120 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/46857#-M4yVK0NDeQlFLlpTC_L:-M4yVK0NDeQlFLlpTC_M:b1eds2d) ([raw file](https://github.com/cockroachdb/cockroach/blob/664ea99a797595987ff7dd9fd5b1cedf51350f17/pkg/ui/src/views/databases/containers/databaseSummary/index.tsx#L120)):*
> ```XML
> // export const isTableFlight = (state: AdminUIState, dbName: string) {
> 
> // }
> ```

Remove commented code please :) 
___
*[pkg/ui/src/views/shared/components/sortabletable/index.tsx, line 293 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/46857#-M4yVoxG2Ib9ljgt0tUy:-M4yVoxG2Ib9ljgt0tUz:bd4lorh) ([raw file](https://github.com/cockroachdb/cockroach/blob/664ea99a797595987ff7dd9fd5b1cedf51350f17/pkg/ui/src/views/shared/components/sortabletable/index.tsx#L293)):*
> ```XML
>           <div className="table__loading">
>             <Spin className="table__loading--spin" indicator={<Icon component={SpinIcon} spin />} />
>             <span className="table__loading--label">Loading tables...</span>
> ```

This text is specific to the database "Tables". Can we extract the component up to the database tables file? Then it can live next to the "no results" component which I think will help readability.


<!-- Sent from Reviewable.io -->

vladlos committed:
ui: Database loading specific text

Removed commented code
Moved specific table loading text to the upper level to be able to reuse it

Release note (ui): database loading state design updates

dhartunian reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> thanks for this last round of changes.

> Reviewed 2 of 2 files at r3, 4 of 4 files at r4.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46857#-:-M5SLgM8CffBDbs1H_Fk:b50tv6j)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @elkmaster)



<!-- Sent from Reviewable.io -->
